### ISSUE TITLE
quoted octal characters are limited to 3 digits

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11478]
Migrated from [rt.perl.org#94252](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=94252) (status was 'resolved')

Searchable as RT94252$

**Comment by p5pRT** [comment]
### From @yecril71pl

#### Created by krzysztof@ne-1-26.stegny.2a.pl



"\\0363" ne "\\363"; the former stands for 2 characters\.  
This is not something I would expect based on perlop\, 
which does not say what an octal character sequence is\.  
It gives just one example instead&#8203;: "\\033" eq "\\c\[";\.  
It is not evident from the given example what the general form of an octal character is; 
one can infer that it is a sequence of octal digits preceded with 0\.

<details><summary>Perl Info</summary>

```

Flags:
    category=docs
    severity=low

This perlbug was built using Perl 5.12.3 - Fri May  6 13:31:41 UTC 2011
It is being executed now by  Perl 5.12.3 - Fri May  6 13:26:30 UTC 2011.

Site configuration information for perl 5.12.3:

Configured by abuild at Fri May  6 13:26:30 UTC 2011.

Summary of my perl5 (revision 5 version 12 subversion 3) configuration:
   
  Platform:
    osname=linux, osvers=2.6.36, archname=x86_64-linux-thread-multi
    uname='linux build33 2.6.36 #1 smp 2011-02-21 10:34:10 +0100 x86_64 x86_64 x86_64 gnulinux '
    config_args='-ds -e -Dprefix=/usr -Dvendorprefix=/usr -Dinstallusrbinperl -Dusethreads -Di_db -Di_dbm -Di_ndbm -Di_gdbm -Duseshrplib=true -Doptimize=-fmessage-length=0 -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -g -Wall -pipe -Accflags=-DPERL_USE_SAFE_PUTENV'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DPERL_USE_SAFE_PUTENV -DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-fmessage-length=0 -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -g -Wall -pipe',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DPERL_USE_SAFE_PUTENV -DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector'
    ccversion='', gccversion='4.5.1 20101208 [gcc-4_5-branch revision 167585]', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib64 -fstack-protector'
    libpth=/lib64 /usr/lib64 /usr/local/lib64
    libs=-lm -ldl -lcrypt -lpthread
    perllibs=-lm -ldl -lcrypt -lpthread
    libc=/lib64/libc-2.11.3.so, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.11.3'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E -Wl,-rpath,/usr/lib/perl5/5.12.3/x86_64-linux-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib64 -fstack-protector'

Locally applied patches:
    


@INC for perl 5.12.3:
    /usr/lib/perl5/site_perl/5.12.3/x86_64-linux-thread-multi
    /usr/lib/perl5/site_perl/5.12.3
    /usr/lib/perl5/vendor_perl/5.12.3/x86_64-linux-thread-multi
    /usr/lib/perl5/vendor_perl/5.12.3
    /usr/lib/perl5/5.12.3/x86_64-linux-thread-multi
    /usr/lib/perl5/5.12.3
    .


Environment for perl 5.12.3:
    HOME=/home/krzysztof
    LANG=pl_PL.UTF-8
    LANGUAGE=
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/krzysztof/bin:/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/X11R6/bin:/usr/games:/usr/lib/mit/bin:/usr/lib/mit/sbin
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From am0c@perl.kr

I tried to summurize the perldoc in two sentences as I understand\.

1\. m\{\\\\\\d\} or m\{\\\\\\d\\d\} will be interpreted as octal value
when perl is not being confused\. For example\, perl will be
confused with perlrebackslash\, as in s\{\(1\)\}\{\\12\};

2\. m\{\\\\\\d\\d\\d\} will be always interperted as octal value
if the first \\d is zero\. if not\, it is not guaranteed to
work as expected\.

Though\, in the 7th note in "Quote and Quote\-like Operators" in perlop\,
the perldoc says&#8203;:

>>  To avoid this warning\, make sure to
>>  pad your octal number with 0's&#8203;: "\\0128"\.

But "\\0128" contain four digits\. And IMHO only this sentence
does not make sense with the context in the perldoc perlop and
perldoc perlrebackslash\.

And it says&#8203;:

>>  \(but best to not use above 077\, see next paragraph\)

I think in this paragraph the documentation should clearly
say what does work\, and what does not work\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @khwilliamson

On 07/07/2011 04&#8203;:47 AM\, am0c am0c wrote&#8203;:
> I tried to summurize the perldoc in two sentences as I understand\.
>
> 1\. m\{\\\\\\d\} or m\{\\\\\\d\\d\} will be interpreted as octal value
> when perl is not being confused\. For example\, perl will be
> confused with perlrebackslash\, as in s\{\(1\)\}\{\\12\};

I don't understand this sentence\.  If you meant some form of the word 
"ambiguous" instead of "confused"\, then this sentence makes more sense 
to me\.  If there are enough capture groups\, a number that could be 
either an octal number or a backreference to a capture group will be 
interpreted as a backreference\.  If not\, it will be interpreted as an 
octal\.  A leading zero forces octal interpretation\.

> 2\. m\{\\\\\\d\\d\\d\} will be always interperted as octal value
> if the first \\d is zero\. if not\, it is not guaranteed to
> work as expected\.

I believe this is an accurate summation
>
> Though\, in the 7th note in "Quote and Quote\-like Operators" in perlop\,
> the perldoc says&#8203;:
>
>>>   To avoid this warning\, make sure to
>>>   pad your octal number with 0's&#8203;: "\\0128"\.
>
> But "\\0128" contain four digits\. And IMHO only this sentence
> does not make sense with the context in the perldoc perlop and
> perldoc perlrebackslash\.

But the fourth digit is not octal\.  By adding the 0\, you expand the 
octal number "12" from 2 to 3 digits\, so that the 8 is unambiguously not 
supposed to be part of it\, and the warning is avoided\.
>
> And it says&#8203;:
>
>>>   \(but best to not use above 077\, see next paragraph\)
>
> I think in this paragraph the documentation should clearly
> say what does work\, and what does not work\.

What would you suggest instead?  The next paragraph goes into some 
detail about why not to use above 077\, but leaves all examples to the 
linked\-to pod\, which does contain this summary&#8203;:

To summarize\, the C\<\\o\{\}> form is always safe to use\, and the other form 
is safe to use for code points through \\077 when you use exactly three 
digits to specify them\.

**Comment by p5pRT** [comment]
### From @cpansprout

Iâve tweaked the perlop documentation slightly with commit 5db3e51\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
